packages/flutter/test/widgets/scroll_view_test.dart

```diff
@@ -1260,6 +1267,28 @@ void main() {
     ), throwsAssertionError);
   });
 
+  testWidgets('ListView.builder asserts on both non-null itemExtent and prototypeItem', (WidgetTester tester) async {
+    expect(() => ListView.builder(
+      itemBuilder: (BuildContext context, int index) {
+        return const SizedBox();
+      },
+      itemExtent: 100,
+      prototypeItem: const SizedBox(),
+    ), throwsAssertionError);
+  });
+
+  testWidgets('ListView.custom asserts on both non-null itemExtent and prototypeItem', (WidgetTester tester) async {
+    expect(() => ListView.custom(
+      childrenDelegate: SliverChildBuilderDelegate(
+        (BuildContext context, int index) {
+          return const SizedBox();
+        },
+      ),
+      itemExtent: 100,
+      prototypeItem: const SizedBox(),
+    ), throwsAssertionError);
+  });
+
   testWidgets('PrimaryScrollController provides fallback ScrollActions', (WidgetTester tester)  async {
     await tester.pumpWidget(
       MaterialApp(
```

Comment:
we should also have a test that makes sure that when you use prototype item, you can jump the scroll controller to a big number and it doesn't build the intervening items. (In other words, a test that the feature really works for what we want it to do!)

Reply:
+1 to adding a test that actually verifies that setting the prototypeItem gives you the desired behavior.

Reply:
@Hixie @goderbauer 
I am not much proficient in widget tests so I may use a little help from you here...

If I understood correctly what @Hixie said, the test would be something like:
 * Create a `ScrollController`
 * Create a `ListView` with a reasonable item count and provide the scroll controller to it
 * Call `ScrollController` `jumpTo` method to a big number
 * Assert that only the "visible" items are rendered and that the items between the initial and final scroll position are note rendered

Is that right? If so, I'm not sure if I know how to properly proceed with the last step. Is it enough to use `expect`, in this case?

Reply:
You could check whether the itemBuilder is only called for the items that are going to be visible after the jump and not for items it jumped over.

Reply:
@goderbauer I checked `list_view_builder_test.dart` and there were similar tests there that I used as a basis. Actually, I duplicated one of them that used `itemExtent` and changed it to `prototypeItem`, is that ok?